### ISSUE TITLE
improved failure detection

### ISSUE THREAD

**Issue by daler** [bioconda/bioconda-recipes#498]
Currently a PR passes if at least one build passes in the matrix of (Linux, OSX) x (py27, py34, py35).

However, when I run a full build locally with `--verbose`, there are a fair number of failures that we might not have known about. For example one of the recipes I had created is failing, and I didn't know until I looked into these details.

Note that every one of these recipes is built by travis on every single PR. So fixing these should speed CI testing substantially.

I would like to propose the the following two changes:

1. we change the nosetests to fail if *any* builds fail
2. In each currently-failing recipe, either fix the issue or explicitly skip the build (like [this example](https://github.com/bioconda/bioconda-recipes/blob/b622a6ac83aa3db43f801d46f1e6a9372dc94148/recipes/snakemake/meta.yaml#L9))

Here are the currently-failing recipes, each with a "cc" at the end based on the output of:

```
git blame recipes/${recipe}/meta.yaml -p  \
| grep "committer " \
| sed "s/committer //g" \
| sort -u
```

fixed in | recipe | reason | git-blame
----------|--------|---------|-----------
49db5fa |`bcbio_monitor`| fails on py34 because gevent 1.0 is not available |  @guillermo-carrasco
49db5fa |`bx-python`| fails on py34 and py35 because one of the imports calls `string.maketrans` which is absent in py3| @chapmanb
 b2e9dd4 | `functools32` | fails on py34 and py35 because module is a backport to py27 | @daler
49db5fa |`geneimpacts`|fails on py34 and py35 because a dictionary in `effect.py` is changed while being iterated over | @brentp, @chapmanb
49db5fa |`hall-lab-svtools`| fails on py34 and py35 due to an encoding issue --- possibly in     `vcf_group_multiline.py`? | @brainstorm
49db5fa |`htseq` | fails on py34 and py35 because py3 doesn't treat `sys.version_info` as an integer | @acaprez
49db5fa |`ipython-cluster-helper`| fails on py34 and py35 because module `slurm` is missing | @chapmanb
 b2e9dd4   |`last`| fails on py27, not sure why| @tomkinsc
49db5fa |`pyutilib`| fails on py35 because it uses `subprocess.mswindows` which was renamed in py35 | @chapmanb
b2e9dd4  | `pyomo` | fails on py35 because it depends on pyutilib | @chapmanb 
49db5fa |`vawk` |fails on py34 and py35 because of py2-style print statement| @chapmanb
b2e9dd4  |`vispr` |fails on py27 on syntax error (I'm not actually sure why)| @johanneskoester
b2e9dd4  |`hifive`| fails on py27 because it looks like Cython is not included as dependency| @daler
b2e9dd4  |`mageck-vispr`| fails on py27 because it depends on Snakemake (which is py3-only)| @johanneskoester

Thoughts?

**Comment by chapmanb** [comment]
Ryan;
Thanks so much for finding these and for the detailed report. Failing on these packages during the build is a great call, I'd love to see that improvement. To work towards it, I added build skips for all of the failing py27-only packages since those don't yet work cleanly on py3:
```
bcbio_monitor
bx-python
geneimpacts
hall-lab-svtools
htseq
ipython-cluster-helper
pyutilib
vawk
```
which leaves only a handful of py27 problem packages before we could enable more rigorous checks. Thanks again for doing this.

**Comment by daler** [comment]
Thanks Brad. I fixed all the others I could find with the exception of LAST (I'll work on that next). I also added which commit each recipe was fixed in for future reference.

**Comment by daler** [comment]
@tomkinsc, I'm just going to leave LAST on py27 skipped -- looks like a py2 compatibility issue.

**Comment by tomkinsc** [comment]
@daler The LAST issue should be resolved in #531.

**Comment by daler** [comment]
Great!
